### DISCUSSION TITLE
Device Guard block Windows Subsystem for Linux

### DISCUSSION THREAD

**Discussion by Acenl12** [HotCakeX/Harden-Windows-Security#138] [General]
### Tools category

Harden Windows Security Script

### Does your system meet the requirements?

- [X] Yes, my system meets the requirements 👍

### Is your Windows installation genuine?

- [X] Yes, I am using genuine Windows installation. 💯

### Please explain the bug

When I try to run a WSL Distro I get the following:

```
Installing, this may take a few minutes...
WslRegisterDistribution failed with error: 0x800711c7
Error: 0x800711c7 Your organization used Device Guard to block this app. Contact your support person for more info.

Press any key to continue...
```
Please help in making WSL distro possible to run again.

**Comment by HotCakeX** [comment]
Hi,
The message "Your organization used Device Guard to block this app" is related to WDAC policy or Smart App Control. Are you using any of them?

Also since this isn't a bug with the module I'm converting this into a discussion.

**Reply by HotCakeX** [reply]
WSL is in this list

https://learn.microsoft.com/en-us/windows/security/application-security/application-control/windows-defender-application-control/design/applications-that-can-bypass-wdac

To run WSL, you might need to turn off Smart App Control. The Harden Windows Security module or script don't deploy User Mode block list of Microsoft. Only Smart App Control deploys it or a custom WDAC policy that you deploy.

**Reply by Acenl12** [reply]
Hi, thanks for the reply. But my point was that after I ran the Device Guard category in your Harden powershell script WSL stopped working,with the error I gave you. I don't know understand why you are talking about WDAC and smart app control.

**Reply by HotCakeX** [reply]
I explained above that why I talked about WADC and Smart App Control. I know your error says "Device Guard" but that actually means WDAC or features that use it such as Smart App Control. It's not the same "Device Guard category" that is in the module or script.

Everything in the [Device Guard category](https://github.com/HotCakeX/Harden-Windows-Security?tab=readme-ov-file#device-guard) is linked to official Microsoft documents. You can see that none of them block WSL.

For you I created a new VM, fully updated it, applied hardening measures, installed WSL and then installed Ubuntu 22 LTS. It's running fine.

![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/a1a11afb-1266-4a31-aed2-4c3c91210c95)

**Reply by HotCakeX** [reply]
Testing with Debian too, working fine, pretty much everything is enabled in Harden Windows Security module (the missing ones are BitLocker related which I didn't use on VM)

![image](https://github.com/HotCakeX/Harden-Windows-Security/assets/118815227/36752eca-32e8-45f8-9d8c-01e3fd13582e)

**Reply by HotCakeX** [reply]
My suggestion is to use [WDACConfig module](https://github.com/HotCakeX/Harden-Windows-Security/wiki/WDACConfig) and [check](https://github.com/HotCakeX/Harden-Windows-Security/wiki/Confirm-WDACConfig) for deployed WDAC policies on your system

```powershell
Install-Module -Name WDACConfig -Force
Confirm-WDACConfig
```

Maybe you used some other tool before to deploy a policy, I'm not sure
